### DISCUSSION TITLE
Conda Environment in Colab

### DISCUSSION THREAD

**Discussion by whoopi24** [sebastian-hofstaetter/teaching#132] [Advanced IR: Exercise Q&A] [answered]
Hi everyone,

we tried to install conda locally (Miniconda) and to use it in Colab, but it seems so that something went wrong or got lost, since every time when we start a new runtime in Colab, we have to install it again and therefore, we also have to re-build the conda environment. That is very tedious and needs a lot of time which may not be necessary - so my question is, how to build a conda environment in Colab such that it permanently exists there ? Is this possible and has anyone of you a short explanation on that ? That would be really great :)

Thank you and best regards,
Marina Sommer

**Comment by ccalvocm** [comment]
Greetings @whoopi24 

You can specify the directory for conda to install to using

`conda install -p path_to_your_dir`

So, you can mount your google drive and conda install there to make it permanent.

**Reply by whoopi24** [reply]
Thanks, I will try that out!
Best regards,
Marina

**Reply by ccalvocm** [reply]
No need to try @whoopi24 , I made a Google Colab for you to test:

https://github.com/ccalvocm/GEE/blob/main/createEnvironment.ipynb

**Reply by Joerg0** [reply]
Hi ccalvocm, 
thanks a lot for your notebook. I followed it quite literally,
still it seems that the activation of the environment doesn't work. The only difference is that I set   "python=3.6" instead of 3.8 but when I start to install the required packages it tells me that I am running python 3.10 which is not compatible.

I did activate the environment, like you described it, but maybe there is the problem.

What's more: If I start a new runtime and want to use conda after mounting the drive and installing & importing condacolab the command
!conda activate /content/drive/MyDrive/test-env
returns the 
CommandNotFoundError: Your shell has not been properly configured to use 'conda activate'.
To initialize your shell, run     $ conda init <SHELL_NAME>


Do you maybe have an idea why it behaves that way?
Thanks in advance!

**Reply by ccalvocm** [reply]
@Joerg0 you have a typo, it's not !conda activate /content/drive/MyDrive/test-env, but !activate /content/drive/MyDrive/test-env as is in the notebook
![screen](https://github.com/sebastian-hofstaetter/teaching/assets/52244870/3c5b0d10-696d-4fcb-887e-ebd847f6f550)

**Reply by Joerg0** [reply]
True. This was badly formulated.
Actually: If I use
!activate /content/drive/MyDrive/test-env
it runs without error and without output.

Still if I want to check which environment is active with
!which python
!python --version
!echo $PYTHONPATH

i get:
/usr/local/bin/python
Python 3.10.10
/env/python

With:
!conda info --envs
I see both environments:

conda environments:
                             /content/drive/MyDrive/test-env
base                     /usr/local

but I can't activate the test-env with python 3.6.
Also if I install some packages which require py3.6 it aborts with an error recurring to py3.10.

So 
!conda activate /content/drive/MyDrive/test-env
was already a tried out workaround.

**Reply by whoopi24** [reply]
Hi everyone who reads this: this issue is still a problem for our group and we would appreciate every form of help! 

Thanks and best regards,
Marina

**Reply by whoopi24** [reply, marked as answer]
The issue has been solved by Joerg0, thank you for trying out so many different approaches!
If you have problems regarding the conda environment in Google Colab, I refer to the very detailed post about the solution of Joerg0: https://github.com/sebastian-hofstaetter/teaching/discussions/137

Good luck to everyone !
Marina
